### DISCUSSION TITLE
How to propagate a gaussian beam in an atmospheric turbulence?

### DISCUSSION THREAD

**Discussion by reemastha706** [AOtools/aotools#96] [Q&A] [answered]
I am trying to propagate a Gaussian beam at a distance of 1km in the atmosphere. 
```
    ```
def Fresnelpropagation(self,num):
        Uin = self.electric_field()  #initially a gaussian beam without affected by turbulence
        phase_arr = self.PS(num)  #gives an array of 5 phase screens
        Uout = []
        Uout.append(Uin)
        for i in range(num):
            Uin = Uin*phase_arr[i]
            Uout.append(opticalpropagation.angularSpectrum(Uin, self.lambda_, self.delta, self.delta, self.z))
        return np.array(Uout)
```


I am not sure what exactly is happenening inside angular spectrum but I supposed the value of z in the function is supposed to be the distance between the two phase screens(if we use multiple), rather than just the final propagation distance which is 1km in my case, if I am not wrong. 

From what I understand, 

source..........A...............B.................C.............D...............receiver plane
where A,B,C,D,E are phase screens. The output of gaussian beam after propagation through A should be the input of the function(Uin) angularSpectrum() and the loop continues. 

Could anyone help me if I am understanding it correct?

Thank you.

**Comment by matthewtownson** [comment, marked as answer]
That is exactly how to use the `angularSpectrum` function, you have understood it perfectly.

One thing I noticed in the snippet you posted is that you multiply the complex field by the wavefront from the `phase_arr` (first line in your `for` loop). I just wanted to make sure that the `phase_arr` is a complex amplitude and not just the phase deviations (as you would get from the phase screen generation tools in aotools). If it is just a phase deviation, you would also need to convert this to an E field using something like:

$U_{\textrm{out}} = A e^{i\theta}$,

where $\theta$ is the phase array, before you mulltiplied it with the existing E field.

**Reply by reemastha706** [reply]
Ah yes, i realized that later. I directly changed the source code for multiple propagation based on the algorithm which was given in the book from Schmidt.

**Comment by matthewtownson** [comment]
If you've got everything you need, can you mark the reply as the answer?
